syntax = "proto3";

service InventoryService {
    rpc GetAllEntries (UserId) returns (EntryList) {}
    rpc GetEntry (EntryId) returns (Entry) {}
    rpc CreateEntry (Entry) returns (Entry) {}
    rpc UpdateEntry (Entry) returns (Empty) {}
    rpc DeleteEntry (EntryId) returns (Empty) {}
    
    rpc GetAllItems (UserId) returns (ItemList) {}
    rpc GetItem (ItemId) returns (Item) {}
    rpc CreateItem (Item) returns (Item) {}
    rpc UpdateItem (Item) returns (Item) {}
    rpc DeleteItem (ItemId) returns (Empty) {}
}

message Empty {}

message UserId {
    string id = 1;
}

message Item {
    string id = 1;
    string name = 2;
    string description = 3;
    string userId = 4;
}

message Entry {
    string id = 1;
    Item item = 2;
    float price = 3;
    int32 amount = 4;
    string date = 5;
    string note = 6;
    string userId = 7;
}

message EntryList{
    repeated Entry entries = 1;
}

message ItemList{
    repeated Item items = 1;
}

message EntryId{
    string id = 1;
}

message ItemId{
    string id = 1;
}